resource-detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'google-cloud-env'
+
+module OpenTelemetry
+  module Resource
+    module Detectors
+      # GoogleCloudPlatform contains detect class method for determining gcp environment resource labels
+      module GoogleCloudPlatform
+        extend self
+
+        def detect # rubocop:disable Metrics/AbcSize
+          gcp_env = Google::Cloud::Env.new
+          resource_labels = {}
+          resource_constants = OpenTelemetry::SDK::Resources::Constants
+
+          if gcp_env.compute_engine?
+            resource_labels[resource_constants::CLOUD_RESOURCE[:provider]] = 'gcp'
+            resource_labels[resource_constants::CLOUD_RESOURCE[:account_id]] = gcp_env.project_id
+            resource_labels[resource_constants::CLOUD_RESOURCE[:region]] = gcp_env.instance_attribute('cluster-location')
+            resource_labels[resource_constants::CLOUD_RESOURCE[:zone]] = gcp_env.instance_zone
+
+            resource_labels[resource_constants::HOST_RESOURCE[:hostname]] = hostname
+            resource_labels[resource_constants::HOST_RESOURCE[:id]] = gcp_env.lookup_metadata('instance', 'id')
+            resource_labels[resource_constants::HOST_RESOURCE[:name]] = gcp_env.lookup_metadata('instance', 'hostname')
```

Comment:
At some point, it might be worth adding a helper method (or methods) to `OpenTelemetry::SDK::Resources`, like:
```ruby
OpenTelemetry::SDK::Resources.host_labels(
  hostname: hostname,
  id: gcp_env.lookup_metadata('instance', 'id'),
  name: gcp_env.lookup_metadata('instance', 'hostname'),
)
```

Reply:
I agree, that would be a good helper.  I don't think it fits any of the use cases that are part of this PR however (I wouldn't use it in the google detector as an example).  Is it something you would like as part of this PR?